src/database.ts

```diff
@@ -13,33 +13,62 @@
 // limitations under the License.
 
 import * as aws from "@pulumi/aws";
+
 import { EnforcementLevel, ResourceValidationPolicy, validateTypedResource } from "@pulumi/policy";
 
-export const database: ResourceValidationPolicy[] = [
-    redshiftClusterConfigurationCheck("mandatory", true /* clusterDbEncrypted */, true /* loggingEnabled */),
-    redshiftClusterMaintenanceSettingsCheck("mandatory", true /* allowVersionUpgrade */),
-    redshiftClusterPublicAccessCheck("mandatory"),
-    dynamodbTableEncryptionEnabled("mandatory"),
-    rdsInstanceBackupEnabled("mandatory"),
-    rdsInstancePublicAccessCheck("mandatory"),
-    rdsStorageEncrypted("mandatory"),
-];
-
-/**
- *
- * @param [enforcementLevel="advisory"] The enforcement level to enforce this policy with.
- * @param [clusterDbEncrypted=true] If true, database encryption is enabled.
- * @param [loggingEnabled=true] If true, audit logging must be enabled.
- * @param [nodeTypes] List of allowed node types.
- */
+import { registerPolicy } from "./awsGuard";
+import { defaultEnforcementLevel } from "./enforcementLevel";
+import { PolicyArgs } from "./policyArgs";
+import { getValueOrDefault } from "./util";
+
+// Mixin additional properties onto AwsGuardArgs.
+declare module "./awsGuard" {
+    interface AwsGuardArgs {
+        redshiftClusterConfigurationCheck?: EnforcementLevel | RedshiftClusterConfigurationCheckArgs;
+        redshiftClusterMaintenanceSettingsCheck?: EnforcementLevel | RedshiftClusterMaintenanceSettingsCheckArgs;
+        redshiftClusterPublicAccessCheck?: EnforcementLevel;
+        dynamodbTableEncryptionEnabled?: EnforcementLevel;
+        rdsInstanceBackupEnabled?: EnforcementLevel | RdsInstanceBackupEnabledArgs;
+        rdsInstancePublicAccessCheck?: EnforcementLevel;
+        rdsStorageEncrypted?: EnforcementLevel | RdsStorageEncryptedArgs;
+    }
+}
+
+// Register policy factories.
+registerPolicy("redshiftClusterConfigurationCheck", redshiftClusterConfigurationCheck);
+registerPolicy("redshiftClusterMaintenanceSettingsCheck", redshiftClusterMaintenanceSettingsCheck);
+registerPolicy("redshiftClusterPublicAccessCheck", redshiftClusterPublicAccessCheck);
+registerPolicy("dynamodbTableEncryptionEnabled", dynamodbTableEncryptionEnabled);
+registerPolicy("rdsInstanceBackupEnabled", rdsInstanceBackupEnabled);
+registerPolicy("rdsInstancePublicAccessCheck", rdsInstancePublicAccessCheck);
+registerPolicy("rdsStorageEncrypted", rdsStorageEncrypted);
+
+export interface RedshiftClusterConfigurationCheckArgs extends PolicyArgs {
+    /** If true, database encryption is enabled. Defaults to true. */
```

Comment:
Does this work for autogenerating docs?

Reply:
Will check. We may need to actually export them from the top-level index. In which case, I don't want them to be alongside `AwsGuard` and `AwsGuardArgs`. The pattern we've used is to export them from a `types/input` module. This way, the types are available in docs, but when you import the top-level module, you don't get tons of noise.

Reply:
I haven't checked the doc generation for this library yet, since we're not going to add docs yet, but I will check subsequently, and make any necessary changes for the docs to be decent for these.